src/mumble/Log.cpp

```diff
@@ -415,20 +415,26 @@ QString Log::validHtml(const QString &html, bool allowReplacement, QTextCursor *
 
 	qtd.adjustSize();
 	QSizeF s = qtd.size();
+	int messageSize = s.width() * s.height();
+	int allowedSize = 2048 * 2048;
+
+	if (messageSize > allowedSize || !s.isValid()) {
```

Comment:
I would prefer it `if (!s.isValid())` was its own section like this.

Something like:

    if (!is.isValid()) {
        QString errorInvalidSizeMessage = tr("[[ Invalid size ]]");
        if (tc) {
           tc->insertText(errorInvalidSizeMessage);
           return QString();
        } else {
           return errorInvalidSizeMessage;
       }
    }

and I'd do the check just after we've retrieved `QSizeF s`.

Then, the section below this would just be...

    int messageSize = s.width() * s.height();
    int allowedSize = 2048 * 2048;

    if (messageSize > allowedSize) {
       [...]
    }